internal/server/add_payment.go

```diff
@@ -0,0 +1,83 @@
+package server
+
+import (
+	"github.com/ministryofjustice/opg-go-common/template"
+	"github.com/ministryofjustice/opg-sirius-lpa-frontend/internal/sirius"
+	"net/http"
+	"regexp"
+	"strconv"
+)
+
+type AddPaymentClient interface {
+	AddPayment(ctx sirius.Context, caseID int, amount int, source string, paymentDate sirius.DateString) error
+	Case(sirius.Context, int) (sirius.Case, error)
+}
+
+type addPaymentData struct {
+	XSRFToken string
+	Success   bool
+	Error     sirius.ValidationError
+
+	Case        sirius.Case
+	Amount      string
```

Comment:
I have tried setting the amount to both `int` (formatted as float in the UI)  and `float64` however both introduced complexity and issues around formatting the amount to and from an int (in pence) on submission, eg on validation error. Setting it to string feels like the smoothest option

Reply:
Ahhh, interesting, that's a good solution